pkg/ignition/network.go

```diff
@@ -0,0 +1,210 @@
+	networkConfigTPlNetworkd = `{{- $element := . -}}
+{{- $type := $element.Type -}}
+{{ if eq $type "ethernet" -}}
+[Match]
+MACAddress={{ $element.MacAddress }}
+
+{{- if .LinkMTU }}
+MTUBytes={{ .LinkMTU }}
+{{- end }}
+
+[Network]
+{{- if .VRF }}
+VRF={{ .VRF }}
+{{- end }}
+{{- if and $element.DHCP4 $element.DHCP6 }}
+DHCP=yes
+{{- else if $element.DHCP4 }}
+DHCP=ipv4
+{{- else if $element.DHCP6 }}
+DHCP=ipv6
+{{- end }}
+
+{{- template "dns" . }}
+
+{{- if $element.IPAddress }}
+[Address]
+Address={{ $element.IPAddress }}
+{{- end }}
+
+{{- if $element.IPV6Address }}
+[Address]
+Address={{ $element.IPV6Address }}
+{{- end }}
+
+{{ if or .Gateway .Gateway6 }}
+{{- if .Gateway -}}
+[Route]
+Destination=0.0.0.0/0
+Gateway={{ .Gateway }}
+{{- if .Metric }}
+Metric={{ .Metric }}
+{{- end }}
+{{- end }}
+
+{{- if .Gateway6 }}
+[Route]
+Destination=::/0
+Gateway={{ .Gateway6 }}
+{{- if .Metric6 }}
+Metric={{ .Metric6 }}
+{{- end }}
+{{- end }}
+{{- end }}
+{{ template "routes" . -}}
+{{ template "rules" . -}}
+{{- end -}}
+
+{{- if eq $type "vrf" -}}
+[Match]
+Name={{ $element.Name }}
+{{ template "routes" . -}}
+{{- template "rules" . -}}
+{{- end -}}
+
+{{- define "dns" }}
+{{- if .DNSServers }}
+{{- range $dnsServer := .DNSServers }}
+DNS={{ $dnsServer }}
+{{- end }}
+{{- end }}
+{{- end }}
+
+{{- define "rules" }}
+{{- if .FIBRules }}
+{{- range $index, $rule := .FIBRules }}
+
+[RoutingPolicyRule]
+{{ if $rule.To }}To="{{$rule.To}}"{{- end }}
+{{ if $rule.From }}From="{{$rule.From}}"{{- end }}
+{{ if $rule.Priority }}Priority={{$rule.Priority}}{{- end }}
+{{ if $rule.Table }}Table={{$rule.Table}}{{- end }}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+
+{{- define "routes" }}
+{{- if .Routes }}
+{{- range $index, $route := .Routes }}
+[Route]
+{{ if $route.To }}Destination={{$route.To}}{{- end }}
+{{ if $route.Via }}Gateway={{$route.Via}}{{- end }}
+{{ if $route.Metric }}Metric={{$route.Metric}}{{- end }}
+{{ if $route.Table }}Table={{$route.Table}}{{- end }}
+{{- end -}}
+{{- end -}}
+{{- end -}}
+`
+
+	netDevConfigTpl = `{{- $element := . -}}
+{{- if eq $element.Type "vrf" -}}
+[NetDev]
+Name={{ $element.Name }}
+Kind={{ $element.Type }}
+
+[VRF]
+Table={{ $element.Table }}
+{{- end }}
+`
+)
+
+// RenderNetworkConfigData renders network-config data into systemd-networkd unit files.
+func RenderNetworkConfigData(data cloudinit.BaseCloudInitData) (map[string][]byte, error) {
+	configs := make(map[string][]byte)
+
+	// adjust VRFs
+	adjustVrfs(data.NetworkConfigData)
+
+	// Add VRFs first so that they are created before the ethernet interfaces.
+	n := 0
+	for i, networkConfig := range data.NetworkConfigData {
+		if networkConfig.Type == networkTypeVRF {
+			config, err := render(fmt.Sprintf("%d-%s", i, networkConfig.Type), netDevConfigTpl, networkConfig)
+			if err != nil {
+				return nil, err
+			}
+
+			name := "00-vrf0.netdev"
+			switch {
+			case i < 10 && networkConfig.Type == networkTypeVRF:
+				name = fmt.Sprintf("%d0-vrf%d.netdev", n, n)
```

Comment:
Shouldn't the `0` come before the number? If so, you could use a `%02d` format and get rid of the switch.

Reply:
it's rare, to have an interface or unit number more then 10, i think it couldn't picked up and the order is wrong

Reply:
Alright, my bad.
I will do that.
 thanks a great catch